src/EPub.ts

```diff
@@ -0,0 +1,244 @@
+      for (let j = 0; j < xml.attributes.length; j++) {
+        const attribute = xml.attributes.item(j);
+        obj["@attributes"][attribute.nodeName] = attribute.nodeValue;
+      }
+    }
+  }
+  // process TEXT_NODE
+  else if (xml.nodeType == 3) {
+    obj = xml.nodeValue;
+  }
+
+  if (xml.hasChildNodes()) {
+    for (let i = 0; i < xml.childNodes.length; i++) {
+      const item = xml.childNodes.item(i);
+      const nodeName: string = item.nodeName;
+
+      if (typeof obj[nodeName] == "undefined") {
+        obj[nodeName] = xmlToJson(item);
+      } else {
+        //@ts-ignore
+        if (typeof obj[nodeName].push == "undefined") {
+          const old = obj[nodeName];
+          obj[nodeName] = [];
+          //@ts-ignore
+          obj[nodeName].push(old);
+        }
+        //@ts-ignore
+        obj[nodeName].push(xmlToJson(item));
+      }
+    }
+  }
+  return obj;
+}
+export interface Metadata {
+  title?: string;
+  author?: string;
+  identifier?: string;
+  language?: string;
+  modified?: string;
+}
+
+export interface Link {
+  rel?: Array<string>;
+  href?: string;
+  type?: string;
+  title?: string;
+  children?: Array<Link>;
+}
+
+export default class Manifest {
+  public readonly metadata: Metadata;
+  public readonly links: Array<Link>;
+  public readonly spine: Array<Link>;
+  public readonly resources: Array<Link>;
+  public readonly toc: Array<Link>;
+  private readonly manifestUrl: URL;
+
+  public static async getManifest(
+    manifestUrl: URL,
+    store?: Store
+  ): Promise<Manifest> {
+    const fetchManifest = async (): Promise<Manifest> => {
+      const manifestJSON = await window
+        .fetch(manifestUrl.href)
+        .then((response) => response.text())
+        .then((str) => new window.DOMParser().parseFromString(str, "text/xml"))
+        .then((data) => JSON.stringify(xmlToJson(data)));
+
+      if (store) {
+        await store.set("manifest", JSON.stringify(manifestJSON));
+      }
+
+      return new Manifest(JSON.parse(manifestJSON), manifestUrl);
+    };
+
+    const tryToUpdateManifestButIgnoreResult = async (): Promise<void> => {
+      try {
+        await fetchManifest();
+      } catch (err) {
+        console.log("Something went wrong", err);
+      }
+      return new Promise<void>((resolve) => resolve());
+    };
+
+    // Respond immediately with the manifest from the store, if possible.
+    if (store) {
+      const manifestString = await store.get("manifest");
+      if (manifestString) {
+        // Kick off a fetch to update the store for next time,
+        // but don't await it.
+        tryToUpdateManifestButIgnoreResult();
+        const manifestJSON = JSON.parse(manifestString);
+        return new Manifest(manifestJSON, manifestUrl);
+      }
+    }
+
+    return fetchManifest();
+  }
+
+  public parseMetaData(manifestJSON: any): any {
+    return manifestJSON?.package?.metadata
+      ? {
+          title: manifestJSON.package.metadata["dc:title"]["#text"],
```

Comment:
This metadata assignment needs to be updated to dynamically parse more robust metadata from the package.opf:  
```

dc:creator | <empty> or <unknown> | author
-- | -- | --
dc:contributor | <empty> or <unknown> | contributor
dc:creator or dc:contributor | aut | author
dc:creator or dc:contributor | pbl | publisher
dc:creator or dc:contributor | trl | translator
dc:creator or dc:contributor | edt | editor
dc:creator or dc:contributor | ill | illustrator
dc:creator or dc:contributor | art | artist
dc:creator or dc:contributor | clr | colorist
dc:creator or dc:contributor | nrt | narrator
dc:publisher | N/A | publisher
media:narrator | N/A | narrator
```
source: https://github.com/readium/architecture/blob/master/streamer/parser/metadata.md 

Does the e-reader only care about a subset of potential metadata?

Reply:
It should parse all of the metadata, if available.  We shouldn't strip it out ...

Reply:
Unless this is used to explicitly populate a title or something ... (Where is it used?)

Reply:
It looks like the only place that `Manifest.metadata` is used is to set the title in `IFrameNavigator.ts#L899`

In that case you might want to replace `Manifest.metadata` with `Manifest.title` for clarity and save some work